python/dlr/dlr_model.py

```diff
@@ -77,26 +57,15 @@ class DLRModelImpl(IDLRModel):
     dev_id : int
         Device ID
     """
+    
+    # TVM crashes with `Check failed: override: Global PackedFunc runtime.module.loadfile_so is already registered`
+    # when we try to link muliple versions of the dlr withing the same process. In order to prevent this we make 
+    # _LIB a class attribute which is instantiated only once per class.
+    _LIB = None 
```

Comment:
Technically it does not matter where we keep `_LIB` on module level or inside the class as a Class Variable
Conceptually `_LIB` is module level attribute because it represent `libdlr.so` (we load single `libdlr.so` for the whole lifecycle of DLR module in Python process)
After that change we have to prefix `_LIB` with class name - `DLRModelImpl`.
There is no big reason to change `_LIB` to be Class Variable. 
Lets look at TVM as an example - in TVM `_LIB` is module level attribute - https://github.com/apache/incubator-tvm/blob/master/topi/python/topi/cpp/impl.py#L42

Reply:
Module level instantiations (like this one https://github.com/apache/incubator-tvm/blob/master/topi/python/topi/cpp/impl.py#L42) is generally a bad practice. Any exception during the instantiation will cause the entire process to crash. Instead of keep then exception at instantiation this bubbles up the error to import time. This leads to non descriptive error messages and bad UX. I would say we should avoid patterns like the one in TVM as well.

Reply:
More over to catch such crashes instead of doing global imports, we start doing local imports which causes memory leaks.

Reply:
Technically it does not matter if we use short name `_LIB` or long name `DLRModelImpl._LIB`. What we are discussing here is variable naming question. Currently DLR and TVM use short name for `_LIB`. This PR puts `_LIB` to `DLRModelImpl` "namespace". As a result we have to use `DLRModelImpl._LIB` instead of short name `_LIB`. Moving `_LIB` to  `DLRModelImpl` "namespace" is not necessary for this particular PR. I'd open a separate PR for variable renaming / moving them to new namespaces.

Reply:
This is not a variable naming question. It's the context in which the variable is being instantiated and used. _LIB is a being instantiated in the class and used within the same class. 

Why _LIB is a class variable?

It depends on the model artifact's path (which is part of instance) to dynamically link the dlr.

Reply:
Just to be clear of what change I'd like to see is to move `_LIB = None` from line 104 to line 86. The code will work the same as before.

Reply:
@apivovarov Lokesh informed me that this particular point has been clarified offline. Let me know if the PR is good to merge. Thanks!

Reply:
Resolved this offline.